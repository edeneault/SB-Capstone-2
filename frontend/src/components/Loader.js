import React from "react";
import { Spinner } from "react-bootstrap";

const Loader = () => {
  return (
    <>
      <Spinner
        animation='grow'
        size='lg'
        variant='dark'
        style={{
          position: "fixed",
          top: "50%",
          left: "50%",
          marginTop: "-50px", //Half of the height of the loading graphic
          marginLeft: "-50px",
        }}
      >
        <span className='sr-only'> Loading...</span>
      </Spinner>
    </>
  );
};

export default Loader;
