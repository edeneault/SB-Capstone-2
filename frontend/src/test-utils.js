import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
