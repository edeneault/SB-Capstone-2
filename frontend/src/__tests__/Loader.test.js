import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "../test-utils";
import { initialState } from "../store-mock";

import Loader from "../components/Loader";

const mockStore = configureMockStore([thunk]);

test("renders Loader component without crashing", () => {
  const store = mockStore(initialState);

  const { getByText } = render(
    <Provider store={store}>
      <Loader />
    </Provider>,
  );

  expect(getByText("Loading...")).toBeInTheDocument();
});

test("matches snapshot", () => {
  const store = mockStore(initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <Loader />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
