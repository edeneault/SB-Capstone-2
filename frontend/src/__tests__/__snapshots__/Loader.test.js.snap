// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-grow spinner-grow-lg text-dark"
    style="position: absolute; top: 50%; left: 50%; margin-top: -50px;"
  >
    <span
      class="sr-only"
    >
       Loading...
    </span>
  </div>
</DocumentFragment>
`;
